app/modules/sightings/resources.py

```diff
@@ -57,21 +80,182 @@ def get(self, args):
+            'encounters' not in data and 'encounters' in result_data
+        ):
+            log.error(
+                'Sighting.post missing encounters in one of %r or %r'
+                % (data, result_data)
+            )
+            _cleanup_post_and_abort(
+                None, None, 'Missing encounters between data and result'
+            )
+        if not len(data['encounters']) == len(result_data['encounters']):
+            log.error(
+                'Sighting.post imbalanced encounters in %r or %r' % (data, result_data)
+            )
+            _cleanup_post_and_abort(
+                None, None, 'Imbalance in encounters between data and result'
+            )
+        # i think this was official declared as law today 2021-02-24
+        if len(data['encounters']) < 1:
+            log.error('Sighting.post empty encounters in %r / %r' % (data, result_data))
+            _cleanup_post_and_abort(None, None, 'Must have at least one encounter')
+
+        # now we handle asset-related json that came in. note: arrays should be parallel in data/result_data
+        # this makes sure assetReferences is well-formed and also maps (future) assets to encounters
+
+        # NOTE for simplicity i am going to assume assetReferences *share a common transaction id*  !!
+        #  this will very likely always be true, but in the event it proves not to be, this will have to be altered
+        #  to do multiple submission.import_tus_files() calls below
+
+        arefs_found = {}
+        i = 0
+        while i < len(data['encounters']):
+            enc_data = data['encounters'][i]
+            # TODO handle regular .assets flavor
+            # TODO genericize this across here and encounters
+            if 'assetReferences' in enc_data and isinstance(
+                enc_data['assetReferences'], list
+            ):
+                enc_data['_paths_wanted'] = []
+                for aref in enc_data['assetReferences']:
+                    if (
+                        not isinstance(aref, dict)
+                        or 'path' not in aref
+                        or 'transactionId' not in aref
+                    ):
+                        log.error(
+                            'Sighting.post malformed assetReferences data: %r' % (aref)
+                        )
+                        _cleanup_post_and_abort(
+                            None, None, 'Malformed assetReferences data'
+                        )
+                    enc_data['_paths_wanted'].append(aref['path'])
+                    key = ':'.join((aref['transactionId'], aref['path']))
+                    if key in arefs_found:
+                        arefs_found[key]['encs'].append(i)
+                    else:
+                        arefs_found[key] = aref.copy()
+                        arefs_found[key]['encs'] = [i]
+            i += 1
+
+        # now we have a mapping of encounters to assetReferences... so we try to create the submission + assets
```

Comment:
Could this be moved into a separate private function just to break this function up a bit?